llvm/lib/Transforms/Scalar/JumpThreading.cpp

```diff
@@ -381,6 +385,26 @@ bool JumpThreadingPass::runImpl(Function &F_, FunctionAnalysisManager *FAM_,
   return EverChanged;
 }
 
+SmallVector<BasicBlock *, 32>
+JumpThreadingPass::unreachableFromBB(BasicBlock *BB, DomTreeUpdater *DTU) {
```

Comment:
This feels like a heavy hammer, than you also shouldn't need to reimplement. Can you just skip any block where !DT.isReachableFromEntry?

Reply:
Let me see if it's possible to make the pass track new unreachable blocks correctly using its `Unreachable` container